src/key-path.js

```diff
@@ -6,9 +6,36 @@ var isFunction = require('./utils').isFunction
  * @param {*} toTest
  * @return {boolean}
  */
-exports.isKeyPath = function(toTest) {
+var isKeyPath = exports.isKeyPath = function isKeyPath(toTest) {
   return (
     isArray(toTest) &&
     !isFunction(toTest[toTest.length - 1])
   )
 }
+
+/**
+ * Determines if two keyPaths are the same
+ * @param {array} keyPath
+ * @param {array} otherKeyPath
+ * @return {boolean}
+ */
+exports.eql = function eq(keyPath, otherKeyPath) {
+  if (!isKeyPath(keyPath) || !isKeyPath(otherKeyPath)) {
+    return false
+  }
+
+  var len
+  var i = len = keyPath.length
+
+  if (len !== otherKeyPath.length) {
+    return false
+  }
+
+  while (i--) {
+    if (keyPath[i] !== otherKeyPath[i]) {
```

Comment:
This needs to be recursive since getters can be composed of other getters.

It may be good to rely on the getetr hashCode, although I am a bit hesitant as others have pointed out that there is a possibility of collisions.